client/src/components/main_section/main_section.coffee

```diff
@@ -12,7 +13,7 @@ SingleRepo = React.createFactory require '../repo_list/single_repo.coffee'
 
 MainSection = React.createClass
   render: ->
-    {repo, repoActions, login, pageLocation, pageLocationActions,loginActions} = @props
+    {repo, repoActions, login, pageLocation, pageLocationActions,loginActions, registerActions} = @props
```

Comment:
Yeah, this component is basically the gateway to a whole bunch of other components, since my page abstraction is weak. If I wrap this in something better that handles what "page" we're on, then this prop catastrophe shouldn't be as bad.

This raises the question of exactly where should be know about redux, and where should we not. I'm leaning toward making the page gateway just know about page location stuff, and then have each of the pages know about just their relevant stuff, direct from the store rather than from props like it is now. I'll look into it more.